root/release/edit/tracklist.tt

```diff
@@ -209,8 +209,9 @@
       <input class="pos" type="text" data-bind="value: number" />
     </td>
 
-    <td class="title">
-      <input type="text" class="track-name" data-bind="value: name, valueUpdate: 'input'" />
+    <td class="title" data-bind="css: { 'preview-differs': previewNameDiffers() }">
+      <input type="text" class="track-name" data-bind="value: name, valueUpdate: 'input', visible: !previewNameDiffers()" />
+      <input type="text" class="preview" readonly data-bind="value: previewName, visible: previewNameDiffers()" />
```

Comment:
The only issue I see with this is that switching the inputs loses focus. There should be a way to do this with only a single input by using a writable computed observable...something like this:

```JS
    this.inputName = ko.computed({
      read: ko.computed(() => {
        const preview = this.previewName();
        return preview === null ? this.name() : preview;
      }),
      write: this.name,
      owner: this,
    });
```

And then:
```HTML
    <td class="title">
      <input type="text" class="track-name" data-bind="value: inputName, valueUpdate: 'input', css: { 'preview': previewNameDiffers() }" />
    </td>
```

So the input will read from `preview` if defined, otherwise `name`; but for writes it only writes to `name`.

One question about the `preview-differs` class: is there a reason we set the background color on the `<td>` and make the input background color transparent rather than just set the background color of the input? I suppose it has something to do with the `/* Override grey background for input[readonly]. */` comment, but I didn't really follow why we can't override that.

Reply:
> I didn't really follow why we can't override that.

I guess my actual question is why there would be a preview for a readonly field.

Reply:
Thanks, the writable-computed approach is much cleaner!

Regarding setting the background color on the `<td>`, it's because it currently has a `padding: 0 5px` that results in weird white space on the sides if only the `<input>`'s background is changed. I'm not sure why it was done that way instead of with padding on the `<input>`s; it looks like it dates back to your initial commit in a5e3363fa15e8a9a13f4984414a544459e18a375 ten years ago. :-) Probably so that other non-input cells like the up and down arrows would get padding?

(The readonly thing is a red herring, I think. I marked the now-unnecessary preview `<input>`s as readonly since they don't accept input, but then I needed to override existing styling for readonly inputs.)

Reply:
I've made some tweaks to `release-editor.less` so that the columns are roughly the same width, but with the padding applied to `<input>`s rather than `<td>`s.